scripts/tasks/dice_roll/task.md

```diff
@@ -0,0 +1,136 @@
+**Linux related topics**: Text processing and filters
+
+---
+## Task:
+Create a script which simulates dice roll with different types and numbers of dice.  
+Follow tasks from number 1 to slowly add functionality into your code.  
+
+For quick reference to specific die we will use NdS notation where N is number of dice  
+and S is number of die sides.  
+e.g. 2d8 means 2 dice with 8 sides
+
+&nbsp;
+
+---
+**EASY:**
+
+**1) Single 1d6**  
+Create the most basic implementation of one die roll. Roll 1d6.
+
+```
+./roll_dice.sh  
+5
+```
+&nbsp;
+
+**2) Multiple 1d6**  
+Make your script roll Nd6 based on your input.
+
+```
+./roll_dice.sh 2  
+3  
+1  
+```
+&nbsp;
+
+**3) Single 1dN. Use die reference in input.**  
+Make your script roll single 1dN based on your input.
+
+```
+./roll_dice.sh 1d12  
+11
+```
+&nbsp;
+
+**4) Multiple 1dN.**  
+Make your script roll single NdN based on your input.
+
+```
+./roll_dice.sh 3d12  
+11  
+7  
+2
+```
+&nbsp;
+
+---
+**AVERAGE:**
+
+**5) Various combinations of NdN.**  
+Make your script roll single NdN based on your input.
+
+```
+./roll_dice.sh 1d12 2d4  
+11  
+3  
+2
+```
+&nbsp;
+
+**6) Add some pretty output**  
+Make your script roll single NdN based on your input. You are not limited in your creativity, add any type of output you like.  
+If you want to make it more complicated, implement option to switch between different output types (human-readable, simple, json format,...)
+
+```
+./roll_dice.sh 1d12 2d4  
+Roll 1d12:  
+11 = 11  
+Roll 2d4:  
+3 + 2 = 5  
+TOTAL: 16
+```
+
+**VERY HARD VARIANT:** JSON variant of output. Hard to implement in shell, you'll have to learn additional linux tools.
+```
+./roll_dice.sh 1d12 2d4  
+ {
+   "results": [
+      {
+        "sides": 12,  
+        "count": 1,  
+        "rolls" [
+            11
+        ]  
+      },
+      {
+        "sides": 4,  
+        "count": 2,  
+        "rolls" [
+            3,
+            2
+        ]  
+      }  
+  ],  
+  "sum": 16  
+}
+```
+&nbsp;
+
+**7) Input check and error handling**  
+Make sure your code handles invalid input args.
+
+```
+./roll_dice.sh abcd  
+ERROR: 'abcd' is not valid input!
+USE: ./roll_dice NdM
+where N (number of dice) and S (number of sides) are positive non-zero numbers
+```
+
+```
+./roll_dice.sh 0d12  
+ERROR: Number of dice or number of sides cannot be 0!
+USE: ./roll_dice NdS
+where N (number of dice) and S (number of sides) are positive non-zero numbers
+```
+&nbsp;
+
+## What to do next?
+If you solved most of defined tasks, you can move to another task or you can improve you solution even more. You can add colors, help option, different output types, export into a file or a permanent logfile which contains all results ever generated by the script with related users and timestamps.
+
+> **ADVICE:**  
+> Don't waste your time to improve this script until you try/finish different tasks.  
+> Your target should be to try multiple different tasks to see different uses for scripts.  
```

Comment:
"The goal should be to tackle as many subtasks as possible to hone your scripting skills gradually."